agent/config/config.go

```diff
@@ -125,6 +125,16 @@ const (
 	ImagePullPreferCachedBehavior
 )
 
+const (
+	// When ContainerInstancePropagateTagsEC2InstanceType is specified, no DescribeTags
+	// API call will be made.
+	ContainerInstancePropagateTagsNoneType ContainerInstancePropagateTagsType = iota
+
+	// When ContainerInstancePropagateTagsEC2InstanceType is specified, agent will
+	// make DescribeTags API call to get tags remotely.
+	ContainerInstancePropagateTagsEC2InstanceType
```

Comment:
Does this get stored in the state file? if so, we should test that the values of these don't change. (IE, we don't accidentally reorder them in the future)